### PR TITLE
Allow to use plugin encoder and decoder

### PR DESCRIPTION
**Pull request by kleag**
This solves issue #2403

### CODE REVIEW

**Comment by vince62s** [comment]
you need to install black version 22 with:
python -m pip install black==22.*
then in openNMT-py folder run:
black .

**Comment by kleag** [comment]
Oups. Sorry. Done.

File: .gitignore [hunk 1]
```diff
@@ -110,3 +110,5 @@ ENV/
 
 # Tensorboard
 runs/
+
+*.kdev4
```
Comment (vince62s) [inline .gitignore:114]: what is this?

File: onmt/bin/train.py [hunk 2]
```diff
@@ -62,7 +62,6 @@ def _get_parser():
 
 def main():
     parser = _get_parser()
-
```
Comment (vince62s) [inline onmt/bin/train.py:65]: try not to modify when not necessary.

**Comment by vince62s** [comment]
are you sure it works as is?
when you add an encoder for instance, it does not make it available in opts so it may fail when you start training.
I think you may need to add dynamically the corresponding opt.

**Pull request merged**